contracts/ForkDelta.sol

```diff
@@ -337,4 +375,78 @@ contract ForkDelta {
     orderFills[msg.sender][hash] = amountGet;
     emit Cancel(tokenGet, amountGet, tokenGive, amountGive, expires, nonce, msg.sender, v, r, s);
   }
+  
+  ////////////////////////////////////////////////////////////////////////////////
+  // Contract Versioning / Migration
+  ////////////////////////////////////////////////////////////////////////////////
+  
+  /**
+  * User triggered function to migrate funds into a new contract to ease updates.
+  * Emits a FundsMigrated event.
+  * @param address Contract address of the new contract we are migrating funds to
+  * @param address[] Array of token addresses that we will be migrating to the new contract
+  */
+  function migrateFunds(address newContract, address[] tokens_) public {
+  
+    require(newContract != address(0));
+    
+    ForkDelta newExchange = ForkDelta(newContract);
+
+    // Move Ether into new exchange.
+    uint etherAmount = tokens[0][msg.sender];
+    if (etherAmount > 0) {
+      tokens[0][msg.sender] = 0;
+      newExchange.depositForUser.value(etherAmount)(msg.sender);
+    }
+
+    // Move Tokens into new exchange.
+    for (n = 0; n < tokens_.length; n++) {
+      address token = tokens_[n];
+      require(token != address(0)); // Ether is handled above.
+      uint tokenAmount = tokens[token][msg.sender];
+      
+      if (tokenAmount == 0) {
+        continue;
+      }
+      
+      require(Token(token).approve(newExchange, tokenAmount));
+      
+      tokens[token][msg.sender] = 0;
+      
+      newExchange.depositTokenForUser(token, tokenAmount, msg.sender);
+    }
+
+    FundsMigrated(msg.sender, newContract);
+  }
+  
+  /**
+  * This function handles deposits of Ether into the contract, but allows specification of a user.
+  * Note: This is intended for use in migration of funds.
```

Comment:
If it is only intended for use from migrateFunds, we could declare the two depositForUser functions internal. That would also lower the attack surface.